test/src/kbasesearchengine/test/parse/ObjectParserTest.java

```diff
@@ -0,0 +1,256 @@
+package kbasesearchengine.test.parse;
```

Comment:
As part of writing the tests, I was thinking that we should be refactoring the code to be understandable, documented, and following good practices such as not returning nulls (and avoiding nulls in general) and that kind of thing.

Reply:
I still think this one needs some work right?